contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,794 @@
+        address USDC;
+        address currentOption;
+        uint256 delay;
+        uint16 lastStrikeOverride;
+        uint256 overriddenStrikePrice;
+    }
+
+    function commitAndClose(
+        address strikeSelection,
+        address optionsPremiumPricer,
+        uint256 premiumDiscount,
+        CloseParams calldata closeParams,
+        Vault.VaultParams calldata vaultParams,
+        Vault.VaultState calldata vaultState,
+        address collateralAsset
+    )
+        external
+        returns (
+            address otokenAddress,
+            uint256 premium,
+            uint256 strikePrice,
+            uint256 delta
+        )
+    {
+        uint256 expiry;
+
+        // uninitialized state
+        if (closeParams.currentOption == address(0)) {
+            expiry = getNextFriday(block.timestamp);
+        } else {
+            expiry = getNextFriday(
+                IOtoken(closeParams.currentOption).expiryTimestamp()
+            );
+        }
+
+        IStrikeSelection selection = IStrikeSelection(strikeSelection);
+
+        (strikePrice, delta) = closeParams.lastStrikeOverride ==
+            vaultState.round
+            ? (closeParams.overriddenStrikePrice, selection.delta())
+            : selection.getStrikePrice(expiry, vaultParams.isPut);
+
+        require(strikePrice != 0, "!strikePrice");
+
+        otokenAddress = getOrDeployOtoken(
+            closeParams.OTOKEN_FACTORY,
+            vaultParams.underlying,
+            closeParams.USDC,
+            collateralAsset,
+            strikePrice,
+            expiry,
+            vaultParams.isPut
+        );
+
+        verifyOtoken(
+            otokenAddress,
+            vaultParams,
+            collateralAsset,
+            closeParams.USDC,
+            closeParams.delay
+        );
+
+        premium = dswmul(
+            GnosisAuction.getOTokenPremium(
+                otokenAddress,
+                optionsPremiumPricer,
+                premiumDiscount
+            ),
+            ISTETH(collateralAsset).pricePerShare().mul(
+                decimalShift(collateralAsset)
+            )
+        );
+
+        require(premium > 0, "!premium");
+    }
+
+    function verifyOtoken(
+        address otokenAddress,
+        Vault.VaultParams calldata vaultParams,
+        address collateralAsset,
+        address USDC,
+        uint256 delay
+    ) private view {
+        require(otokenAddress != address(0), "!otokenAddress");
+
+        IOtoken otoken = IOtoken(otokenAddress);
+        require(otoken.isPut() == vaultParams.isPut, "Type mismatch");
+        require(
+            otoken.underlyingAsset() == vaultParams.underlying,
+            "Wrong underlyingAsset"
+        );
+        require(
+            otoken.collateralAsset() == collateralAsset,
+            "Wrong collateralAsset"
+        );
+
+        // we just assume all options use USDC as the strike
+        require(otoken.strikeAsset() == USDC, "strikeAsset != USDC");
```

Comment:
We removed these checks entirely for VaultLifecycle.sol, feels like these checks are redundant because we're already passing them in as arguments for `factory.createOtoken`

Reply:
I initially thought they are redundant, but actually opyn can change code for token blueprint to create a different token. less attack surface this way

Reply:
Okay, let's do a follow-up PR to add those back into VaultLifecycle.sol